backend/src/db/providers/courseProvider.ts

```diff
@@ -169,15 +169,24 @@ const getCourseByIdProvider = async (
   return convertToTypeUnsafe<ICoursePopulated>(course);
 };
 
-const getMaterialsProvider = async (courseId: string) => {
+const getMaterialsProvider = async (
+  courseId: string,
+): Promise<Pick<ICourse, '_id' | 'materials'>> => {
```

Comment:
Looks terrible. It should be extracted or just leave `ICourse` as return value type